packages/core/src/controllers/relayer.ts

```diff
@@ -400,6 +322,78 @@ export class Relayer extends IRelayer {
   }
 
   // ---------- Private ----------------------------------------------- //
+
+  private async connect(relayUrl?: string) {
+    await this.confirmOnlineStateOrThrow();
+    if (relayUrl && relayUrl !== this.relayUrl) {
+      this.relayUrl = relayUrl;
+      await this.transportDisconnect();
+    }
+
+    this.connectionAttemptInProgress = true;
+    this.transportExplicitlyClosed = false;
+    let attempt = 1;
+
+    while (attempt < 6) {
+      try {
+        this.logger.debug({}, `Connecting to ${this.relayUrl}, attempt: ${attempt}...`);
+        // Always create new socket instance when trying to connect because if the socket was dropped due to `socket hang up` exception
+        // It wont be able to reconnect
+        await this.createProvider();
+
+        await new Promise<void>(async (resolve, reject) => {
+          const onDisconnect = () => {
+            reject(new Error(`Connection interrupted while trying to subscribe`));
+          };
+          this.provider.once(RELAYER_PROVIDER_EVENTS.disconnect, onDisconnect);
+
+          await createExpiringPromise(
+            this.provider.connect(),
+            15_000,
```

Comment:
why `15_000` and not `15000`? I remember having some issues in RN with numbers using underscore

Reply:
updated